### PR TITLE
cmd, eth, internal, les: add gasprice cap

### PR DESCRIPTION
**Pull request by rjl493456442**
This PR adds a CLI flag to cap the maximum gas price.

### CODE REVIEW

**Comment by ppratscher** [comment]
Isn't the default value fo 1000 ETH as max fee a little bit high? I suggest to reduce it to 5 or 10 ETH.

**Comment by rjl493456442** [comment]
@ppratscher It's 1000GWei(price). It's an arbitrary number to be honest. Nowadays the average gas price is ~50GWei. And I notice sometimes the highest gas price is more than 9kGWei. So I pick it as the default value.

**Comment by ppratscher** [comment]
Ah sorry, I misread the code and didn't realize that you just check the gas price and not the max tx fee. I believe it is better to put a cap on the max tx fee and not only on the gas price as it will be simpler to understand for the end user.

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -478,6 +478,11 @@ var (
 		Name:  "rpc.gascap",
 		Usage: "Sets a cap on gas that can be used in eth_call/estimateGas",
 	}
+	RPCGlobalTxFeeCap = cli.Float64Flag{
+		Name:  "rpc.txfeecap",
+		Usage: "Sets a cap on transaction fee(in ether) that can be used in sendTransaction/sendRawTransaction(zero for disable)",
```
Comment (karalabe) [inline cmd/utils/flags.go:483]: I'd rephrase a bit to make it shorter.

```
Sets a cap on transaction fee (in ether) that can be sent via the RPC APIs (0 = no cap)
```

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -1520,6 +1520,13 @@ func (args *SendTxArgs) toTransaction() *types.Transaction {
 
 // SubmitTransaction is a helper function that submits tx to txPool and logs a message.
 func SubmitTransaction(ctx context.Context, b Backend, tx *types.Transaction) (common.Hash, error) {
+	// If the transaction fee cap is already specified, ensure the
+	// fee of the given transaction is _reasonable_.
+	feeEth := new(big.Float).Quo(new(big.Float).SetInt(new(big.Int).Mul(tx.GasPrice(), new(big.Int).SetUint64(tx.Gas()))), new(big.Float).SetInt(big.NewInt(params.Ether)))
+	feeFloat, _ := feeEth.Float64()
+	if b.RPCTxFeeCap() != 0 && feeFloat > b.RPCTxFeeCap() {
+		return common.Hash{}, fmt.Errorf("txfee(%.2f ether) exceeds the maximum setting(%.2f ether)", feeFloat, b.RPCTxFeeCap())
```
Comment (karalabe) [inline internal/ethapi/api.go:1528]: I've noticed that throughout your PRs you started omitting a space character from front of parantheses. Please add that (i.e. `tx fee<space>(%.2f ether) exceeds...`)

Comment (karalabe) [reply internal/ethapi/api.go:1528]: Also instead of `maximum setting` let's write `configured cap`.

Comment (rjl493456442) [reply internal/ethapi/api.go]: Done

File: cmd/utils/flags.go (hunk 1 above)
Comment (rjl493456442) [reply cmd/utils/flags.go]: Done
